src/controllers/authController.js

```diff
@@ -0,0 +1,176 @@
+const User = require('../models/user.js');
+const userService = require('../services/userService.js');
+const tokenService = require('../services/tokenService.js');
+const jwtService = require('../services/jwtService.js');
+const ApiError = require('../exeptions/apiError.js');
+const bcrypt = require('bcrypt');
+
+const validateEmail = (value) => {
+  if (!value) {
+    return 'Email is required';
+  }
+
+  const emailPattern = /^[\w.+-]+@([\w-]+\.){1,3}[\w-]{2,}$/;
+
+  if (!emailPattern.test(value)) {
+    return 'Email is not valid';
+  }
+};
+
+const validaPassword = (value) => {
+  if (!value) {
+    return 'Password is required';
+  }
+
+  if (value.length < 6) {
+    return 'At least 6 characters';
+  }
+};
+
+const register = async (req, res) => {
+  const { name, email, password } = req.body;
+
+  const errors = {
+    email: validateEmail(email),
+    password: validaPassword(password),
+  };
+
+  if (errors.email || errors.password) {
+    throw ApiError.badRequest('Bad request', errors);
+  }
+
+  const hashedPass = await bcrypt.hash(password, 10);
+
+  await userService.register(name, email, hashedPass);
+
+  res.send({ message: 'OK' });
+};
+
+const activate = async (req, res) => {
+  const { activationToken } = req.params;
+
+  const user = await User.findOne({ where: { activationToken } });
+
+  if (!user) {
+    res.sendStatus(404);
+  }
+
+  user.activationToken = null;
+  user.save();
+
+  res.send(user);
+};
+
+const login = async (req, res) => {
+  const { email, password } = req.body;
+
+  const user = await userService.findByEmail(email);
+
+  if (!user) {
+    throw ApiError.badRequest('No such user');
+  }
+
+  const isPasswordValid = bcrypt.compare(password, user.password);
```

Comment:
The `bcrypt.compare` function is asynchronous and returns a promise. You should await this call to ensure it resolves before proceeding.